src/server/aws_lambda.ts

```diff
@@ -49,21 +53,40 @@ function syncHandler<TEvent = any, TResult = any>(handler: SyncHandler<TEvent, T
     return function wrappedLambdaHandler(event, context, cb) {
         hb.__setStore(AsyncStore)
         AsyncStore.run({context: {}, breadcrumbs: []}, () => {
+            const timeoutHandler = setupTimeoutWarning(hb, context)
             try {
                 handler(event, context, (error, result) => {
+                    clearTimeout(timeoutHandler)
                     if (error) {
                         return reportToHoneybadger(hb, error, cb)
                     }
 
                     cb(null, result)
                 });
             } catch (err) {
+                clearTimeout(timeoutHandler)
                 reportToHoneybadger(hb, err, cb)
             }
         })
     }
 }
 
+function shouldReportTimeoutWarning(hb: Honeybadger, context: Context): boolean {
+    return typeof context.getRemainingTimeInMillis === 'function' && !!((hb.config as ServerlessConfig).reportTimeoutWarning)
+}
+
+function setupTimeoutWarning(hb: Honeybadger, context: Context) {
+    if (!shouldReportTimeoutWarning(hb, context)) {
+        return
+    }
+
+    const delay = context.getRemainingTimeInMillis() - ((hb.config as ServerlessConfig).timeoutWarningThresholdMs)
+    return setTimeout(() => {
+        hb.notify(`${context.functionName}[${context.functionVersion}] may timeout`)
+    }, delay > 0 ? delay : 0)
```

Comment:
I'm thinking about this. If the `timeoutWarningThresholdMs` is set to 50ms, does that mean we'll be triggering a callback possibly every 50ms? Wouldn't that be too much of an impact on the function's runtime/performance?

Maybe we should decouple the callback interval from the threshold? We could check, say, every second or 3/4 second. What do you think?

Reply:
I'm not sure I'm following.
The logic is:
- Check the remaining time left for the lambda function and remove `timeoutWarningThresholdMs` (default 50ms) = `delay`
- Schedule a callback with the `delay` to fire and notify Honeybadger

Example:
- Function has `getRemainingTimeInMillis = 3000` (3 seconds)
- Schedule a callback to fire in 2950ms and notify Honeybadger
- When [If] the function resolves, we cancel the callback with `clearTimeout()`

Does this make sense?

Reply:
Oh, my bad. I read that as setInterval() for some reason. Cool.